e2e/cypress/integration/messaging/pinned_posts_spec.js

```diff
@@ -48,4 +49,146 @@ describe('Messaging', () => {
             cy.get(`#rhsPostMessageText_${postId}`).findByText('Pinned').should('not.exist');
         });
     });
+
+    it('MM-T2168 Un-pin a post, disappears from pinned posts RHS', () => {
+        // # Post a message
+        cy.postMessage('This is a post that is going to be pinned then removed.');
+
+        cy.getLastPostId().then((postId) => {
+            // # On a message in center channel, click then pin the post to the channel
+            cy.clickPostPinIcon(postId);
+
+            // # Find the 'Pinned' span in the post pre-header to verify that the post was actually pinned
+            cy.get(`#post_${postId}`).findByText('Pinned').should('exist');
+
+            // # Click pin icon next to search box
+            cy.get('#channelHeaderPinButton').should('exist').click();
+
+            // # View pinned posts RHS
+            cy.get(`#rhsPostMessageText_${postId}`).should('exist');
+
+            // # On a message in center channel, Click [...] > Un-pin from channel
+            cy.getPostMenu(postId, 'Unpin from Channel').click();
+
+            // * Post disappears from RHS
+            cy.get(`#rhsPostMessageText_${postId}`).should('not.exist');
+
+            // * Pinned badge is removed from post in center
+            cy.get(`#post_${postId}`).findByText('Pinned').should('not.exist');
+        });
+    });
+
+    it('MM-T2169 Un-pinning a post in center also removes badge from *search results* RHS', () => {
+        // # Post a message
+        cy.postMessage('Hello');
+
+        cy.getLastPostId().then((postId) => {
+            // # On a message in center channel, click then pin the post to the channel
+            cy.clickPostPinIcon(postId);
+
+            // # Search for "Hello"
+            cy.get('#searchBox').should('be.visible').type('Hello').type('{enter}');
+
+            // * Post appears in RHS search results, displays Pinned badge
+            cy.get(`#searchResult_${postId}`).findByText('Pinned').should('exist');
+
+            // # On a message in center channel, Click [...] > Un-pin from channel
+            cy.getPostMenu(postId, 'Unpin from Channel').click();
+
+            // * Post still appears in RHS search results, but Pinned badge is removed
+            cy.get(`#searchResult_${postId}`).findByText('Pinned').should('not.exist');
+        });
+    });
+
+    it('MM-T2170 Un-pinning a post in *permalink* view also removes badge from saved posts RHS', () => {
+        // # Post a message
+        cy.postMessage('Permalink post.');
+
+        cy.getLastPostId().then((postId) => {
+            // # On a message in center channel, click then pin the post to the channel
+            cy.clickPostPinIcon(postId);
+
+            // # Click save icon
+            cy.clickPostSaveIcon(postId);
+
+            // # In RHS, click Jump to view permalink view
+            cy.get('#channelHeaderFlagButton').should('exist').click();
```

Comment:
Sorry @isacikgoz - is the button still referred to as `FlagButton`?

Reply:
Yes, it's the is of the element in the code :)

Reply:
Cool, thanks, just wanted to be sure :)